engine/runtime/src/main/java/org/enso/interpreter/node/controlflow/caseexpr/ConstructorBranchNode.java

```diff
@@ -36,15 +40,38 @@ public static ConstructorBranchNode build(
 
   @Specialization
   void doAtom(
-      VirtualFrame frame,
-      Object state,
-      Atom target,
-      @CachedLibrary(limit = "10") StructsLibrary structs) {
+      VirtualFrame frame, Object state, Atom target, @Cached FieldsAsArrayNode toArrayNode) {
     if (profile.profile(matcher == target.getConstructor())) {
-      accept(frame, state, structs.getFields(target));
+      var arr = toArrayNode.executeFields(target);
```

Comment:
The benchmarks regressed a bit after removal of `getFields`, but introduction of `FieldsAsArrayNode` that provides a polymorphic cache for each `AtomConstructor` (up to limit 3 - usually there is only one anyway) and explodes the loop that reads the fields one by one seems to deliver the same results.

![Example of AtomBenchmark](https://github.com/enso-org/enso/assets/26887752/3c0efa0f-2ade-4106-bede-a454ad94cd6a)